### ISSUE TITLE
testRequestFailure and testPesistentServiceConnection failing

### ISSUE THREAD

**Issue by drigz** [rosjava/rosjava_core#267]
c746aed176f2915f3c8c53eaee5fabe006c56e3f  (#242) seems to have broken testRequestFailure and testPesistentServiceConnection. The stacktrace in both cases is similar:

```
java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1438)
	at java.util.HashMap$EntryIterator.next(HashMap.java:1472)
	at java.util.HashMap$EntryIterator.next(HashMap.java:1470)
	at com.google.common.collect.AbstractMultimap$EntryIterator.findValueIteratorAndKey(AbstractMultimap.java:1133)
	at com.google.common.collect.AbstractMultimap$EntryIterator.next(AbstractMultimap.java:1147)
	at com.google.common.collect.AbstractMultimap$EntryIterator.next(AbstractMultimap.java:1117)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at org.ros.node.DefaultNodeMainExecutor.shutdown(DefaultNodeMainExecutor.java:161)
	at org.ros.RosTest.tearDown(RosTest.java:54)
```

**Comment by dambrosio** [comment]
@drigz  I am seeing this as well!

**Comment by jubeira** [comment]
Closed via #296.
